src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs

```diff
@@ -134,10 +136,15 @@ public static AttributeCollection FromExisting(AttributeCollection existing, par
         /// <summary>
         /// Gets the attribute with the specified type.
         /// </summary>
-        public virtual Attribute this[Type attributeType]
+        public virtual Attribute? this[Type attributeType]
         {
             get
             {
+                if (attributeType == null)
+                {
+                    throw new ArgumentNullException(nameof(attributeType));
```

Comment:
previously threw `IndexOutOfRangeException` as we found an index in `_foundAttributeTypes` (which was initialised to all `null`) but this was out of the range of the empty collection.